### DISCUSSION TITLE
Upload to Amazon

### DISCUSSION THREAD

**Discussion by esbezr** [saleweaver/python-amazon-sp-api#856] [Q&A] [answered]
Has anyone uploaded documents? Im not exactly sure what arguments to provide to direct the document to where i want it to go. Thanks. All i have currently is:

from sp_api.api import Upload
up = Upload().upload_document()

**Comment by denisneuf** [comment, marked as answer]
Well I used this sub-api to upload documents for Messaging the buyers, so here is an example you can take the idea:

c_file = “files/manual.pdf"
c_type = "application/pdf”
order = “408-1234567-1234567"
path = "messaging/v1/orders/" + order + "/messages/legalDisclosure"

try:

    res = Upload(account=store, marketplace=marketplace).upload_document(
        resource=path,
        file=c_file,
        content_type=c_type
    )

    result = res.payload
    print(result)
    print(result.get("headers"))
    print(result.get("url"))

    # url = result.get("url")

except SellingApiException as ex:
    logging.info(ex)

**Reply by esbezr** [reply]
how did you determine the path? im trying to upload documents to the part of seller central that says "add products via upload". The c_file should just be the file path on my pc for the document and the c_type should be text for a text document right? If you can link where you got this information from i would appreciate that. I also appreciate the quick answer and this has helped alot already. Thanks

**Comment by denisneuf** [comment]
I think you misunderstand the Upload API:

https://developer-docs.amazon.com/sp-api/docs/uploads-api-reference <https://developer-docs.amazon.com/sp-api/docs/uploads-api-reference>

The Uploads API lets you upload files that you can programmatically access using other Selling Partner APIs, such as the A+ Content API and the Messaging API.

So it means you can upload documents to Message the buyers (and attach this documents to the message) or the Files (aka Images) to create A+ content for listings.

As I understand you would like to create products (listings) with the Upload api?

It is not possible with the Upload Api.

**Reply by esbezr** [reply]
ok. do you know if theres another api call i can do that would let me add documents to seller central?

**Comment by denisneuf** [comment]
I am sorry but “ add documents to seller central?” do not let me understand what do you want to do?

There is hundred places in seller central where you can upload documents. What type of document, format, purposes,…in that way I may help you but without information we are wasting our time.

**Reply by esbezr** [reply]
I want to write an api call in python to add .txt documents from a folder on my pc everyday to upload into seller central. Currently im manually adding the file everyday into seller central through the link that says "add products via upload". I would like to automate this process but i dont know if there is an api call to do this. Sorry for not being specific enough the first time.

**Comment by denisneuf** [comment]
The point is why you upload this txt file? To add products, to change the price of the products,…etc 

I suggest you read this endpoint if matches your case:

https://developer-docs.amazon.com/sp-api/docs/feed-type-values <https://developer-docs.amazon.com/sp-api/docs/feed-type-values>

**Reply by esbezr** [reply]
Feeds is what I actually needed. Thank you.
